src/display/editor/outliner.js

```diff
@@ -260,13 +262,37 @@ class Outliner {
 }
 
 class Outline {
+  /**
+   * @returns {string} The SVG path of the outline.
+   */
   toSVGPath() {
     throw new Error("Abstract method `toSVGPath` must be implemented.");
   }
 
+  /**
+   * @returns {Object|null} The bounding box of the outline.
+   */
   get box() {
     throw new Error("Abstract getter `box` must be implemented.");
   }
+
+  /**
+   * @param {number} x - The x coordinate of the box.
+   * @param {number} y - The y coordinate of the box.
+   * @param {number} width - The width of the box.
+   * @param {number} height - The height of the box.
+   * @returns {Object} The outline in the page coordinate system.
+   */
+  serialize(_x, _y, _width, _height) {
+    throw new Error("Abstract method `serialize` must be implemented.");
+  }
+
+  /**
+   * @returns {boolean} true if the outline is free, false otherwise.
+   */
+  get free() {
+    throw new Error("Abstract method `isFree` must be implemented.");
```

Comment:
Can't we just do the following, and thus avoid having to implement this getter in the extending classes?
```suggestion
    return shadow(this, "free", this instanceof FreeHighlightOutline);
```